internal/uvm/scsi.go

```diff
@@ -235,25 +241,55 @@ func (uvm *UtilityVM) AddSCSI(ctx context.Context, hostPath string, uvmPath stri
 //
 // `readOnly` set to `true` if the physical disk should be attached read only.
 func (uvm *UtilityVM) AddSCSIPhysicalDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (*SCSIMount, error) {
-	return uvm.addSCSIActual(ctx, hostPath, uvmPath, "PassThru", readOnly, VMAccessTypeIndividual)
+	return uvm.addSCSIActual(ctx, hostPath, uvmPath, "PassThru", readOnly, VMAccessTypeIndividual, "")
 }
 
-// addSCSIActual is the implementation behind the external functions AddSCSI and
-// AddSCSIPhysicalDisk.
+// AddSCSIExtensibleVirtualDisk adds an extensible virtual disk as a SCSI mount
+// to the utility VM at the next available location. All such disks which are not actual virtual disks
+// but provide the same SCSI interface are added to the UVM as Extensible Virtual disks.
+// As of now only the virtual disks backed by storage spaces are supported.
+//
+// `hostPath` is required. Depending on the type of the extensible virtual disk the format of `hostPath` can
+// be different.
+// In case of storage space disks the host path must be in the
+// `space://{storage_pool_unique_ID}{virtual_disk_unique_ID}` format.
+//
+// `uvmPath` must be provided in order to be able to use this disk in a container.
+//
+// `readOnly` set to `true` if the virtual disk should be attached read only.
+//
+// `vmAccess` indicates what access to grant the vm for the hostpath
+func (uvm *UtilityVM) AddSCSIExtensibleVirtualDisk(ctx context.Context, hostPath, uvmPath string, readOnly bool) (*SCSIMount, error) {
+	if uvmPath == "" {
+		return nil, fmt.Errorf("uvmPath can not be empty for extensible virtual disk")
+	}
+	// check that the EVD is one of the supported types
+	evdType, mountPath, err := ValidateExtensibleVirtualDiskMount(hostPath)
+	if err != nil {
+		return nil, err
+	}
+	return uvm.addSCSIActual(ctx, mountPath, uvmPath, "ExtensibleVirtualDisk", readOnly, VMAccessTypeIndividual, evdType)
+}
+
+// addSCSIActual is the implementation behind the external functions AddSCSI,
+// AddSCSIPhysicalDisk, AddSCSIExtensibleVirtualDisk.
 //
 // We are in control of everything ourselves. Hence we have ref- counting and
 // so-on tracking what SCSI locations are available or used.
 //
 // `attachmentType` is required and `must` be `VirtualDisk` for vhd/vhdx
-// attachments and `PassThru` for physical disk.
+// attachments, `PassThru` for physical disk and "ExtensibleVirtualDisk" for Extensible virtual disk.
```

Comment:
Is Extensible Virtual Disk an established term or something we are now introducing?
I kind of like it but it could be confusing given VirtualDisk is different. I've never heard EVD used before and not sure a storage space is extensible.

Reply:
We have never used `ExtensibleVirtualDisk`s in containerplat before. However, looks like it is well defined for HCS. We can use a different name to refer to it in the containerplat code but when sending requests to the HCS it has to be `ExtensibleVirtualDisk`.

Reply:
Ok that addresses my concern. We aren't defining a new term and it is definitely desirable to stay consistent with what is currently being used. Thanks.